activerecord/lib/active_record/insert_all.rb

```diff
@@ -195,7 +195,7 @@ def extract_types_from_columns_on(table_name, keys:)
           end
 
           def format_columns(columns)
-            quote_columns(columns).join(",")
+            columns.is_a?(Enumerable) ? quote_columns(columns).join(",") : columns
```

Comment:
We generally check for the method we intend to call rather than the type:

```suggestion
            columns.respond_to?(:map) ? quote_columns(columns).join(",") : columns
```